chapters/api_concepts.txt

```diff
@@ -456,6 +456,42 @@ example are the vertex attributes of the
 start with `vertex.` followed by the attribute name.
 ====
 
+Arrays define a valid region of elements, which by default is always the full
+capacity of the array. Implementations can allow applications to keep a singular
+allocation of data, but use a parameter to tell the application only to use a
+subset of elements. This gives applications the ability to more efficiently
+change the number of elements used by parent objects at the cost of memory size
+efficiency.
+
+Implementations provide this functionality by implementing
+<<core_extensions_array1d_region, extension `KHR_ARRAY1D_REGION`>>, which adds
+the following parameter to `ANARIArray1D`:
+
+.Additional parameters accepted by `ANARIArray1D` with <<core_extensions_array1d_region, extension `KHR_ARRAY1D_REGION`>>.
+[cols="<,<,>,<4",options="header,unbreakable"]
+|================================================================
+| Name     | Type         | Description
+| region   |`UINT64_BOX1` | region of elements currently in use
+|================================================================
+
+The element from `region.lower` is _inclusive_, while `region.upper` is
+_exclusive_. This is intended to mimic C++ begin/end iterators on containers.
+
+If `region` is not set, all elements are used as specified when the array was
+constructed. When `region` is set, it is clamped to the range of the array's
+constructed size respectively and warnings should be emitted by the debug layer
+if `region.upper` is larger than capacity.
+
+It is undefined behavior for `region.lower` >= `region.upper`, meaning
```

Comment:
Can we require this stronger? For arrays without this region extension the spec reads "number of elements `numItemsN` must be positive". This would mean `lower` >= `clamped_upper`  is an error (and not undefined behavior).

Alternatively we can also define that
 1. `upper` is clamped from below by `lower+1`
 2. `upper` is clamped from above by "capacity" (like written now)
 3. `lower` is clamped from above by `upper-1`

which would ensure that always `lower<upper` and `size>=1`.

Reply:
> For arrays without this region extension the spec reads "number of elements numItemsN must be positive"

We should probably reword this to be "number of elements must be >= 1", because the size parameter is unsigned so it will always be positive (by some definition of "positive").

I like the clamping idea. Let's go with that!

Reply:
Cool, works for me.